kore/src/Kore/Log/WarnRetrySolverQuery.hs

```diff
@@ -0,0 +1,68 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+
+module Kore.Log.WarnRetrySolverQuery
+    ( WarnRetrySolverQuery
+    , warnRetrySolverQuery
+    ) where
+
+import Prelude.Kore
+
+import Kore.Internal.Predicate
+    ( Predicate
+    )
+import qualified Kore.Internal.Predicate as Predicate
+import Kore.Internal.Variable
+    ( InternalVariable
+    , VariableName
+    , toVariableName
+    )
+import Kore.Unparser
+    ( unparse
+    )
+import Log
+import Pretty
+    ( Pretty (..)
+    )
+import qualified Pretty
+
+newtype WarnRetrySolverQuery =
+    WarnRetrySolverQuery
+        { predicates :: NonEmpty (Predicate VariableName) }
+    deriving (Show)
+
+instance Pretty WarnRetrySolverQuery where
+    pretty WarnRetrySolverQuery { predicates } =
+        Pretty.vsep $
+        [ "The SMT solver failed to solve the following query:"
+        , Pretty.indent 2 "Decide predicate:"
+        , Pretty.indent 4 (unparse predicate)
+        , Pretty.indent 2 "with side conditions:"
+        ]
+        <> fmap (Pretty.indent 4 . unparse) sideConditions
+        <> ["The SMT solver will be reset and the query\
+            \ will be tried one more time."
+           ]
```

Comment:
Let's change the wording here, now that the warning is only shown after the second query succeeds (otherwise an error is thrown).